### ISSUE TITLE
Lagging/Stuttering

### ISSUE THREAD

**Issue by TheVicBro** [tomcucinotta/MuseScore#36]
I recently gotten everything to work, but I noticed that the program seems to lag a lot (I'm running the Windows build) and it makes it so I have to wait for the next LEDs to light up and for my input to be received. 

The issue seems to worsen upon turning on "Highlight mistakes", pressing several wrong keys or many in succession causes the program to freeze up and the audio plays all at once when these stutters happen. I notice this stutter happens when I first click the Play button, I just hear several piano notes all spammed at the same time before it plays the first note and lights up the corresponding LEDs.

**Comment by tomcucinotta** [comment]
Good catch, @TheVicBro. This is among the annoyances that came after the MuseScore 3.x port. The problem was not there with MuseScore 2.x. However, the patch had also fewer features. I'm living with this burden, and the workaround I use is quite simple: speed-up the playback speed to the maximum while tutoring. This way, it is almost certain that the tutor has reached the note to be pressed before you hit the key.
In case you're a devel and willing to help, I'd certainly appreciate an attempt to debug, troubleshoot and come up with a fix to this.
Clearly, rebasing on a more recent MuseScore 3.x code base, or even a 4.x one, would also be useful things to try.
